examples/energy-management-app/esp32/main/main.cpp

```diff
@@ -116,23 +119,48 @@ chip::Credentials::DeviceAttestationCredentialsProvider * get_dac_provider(void)
 
 } // namespace
 
+EVSEManufacturer * EnergyEvse::GetEvseManufacturer()
+{
+    return gEvseManufacturer;
+}
+
 void ApplicationInit()
 {
-    if ((gDelegate == nullptr) && (gInstance == nullptr) && (gEvseManufacturer == nullptr))
+    if ((gDEMDelegate == nullptr) && (gDEMInstance == nullptr))
+    {
+        gDEMDelegate = new DeviceEnergyManagementDelegate();
+        if (gDEMDelegate != nullptr)
+        {
+            gDEMInstance = new DeviceEnergyManagementManager(
+                EndpointId(ENERGY_EVSE_ENDPOINT), *gDEMDelegate,
+                BitMask<DeviceEnergyManagement::Feature, uint32_t>(
+                    DeviceEnergyManagement::Feature::kPowerAdjustment, DeviceEnergyManagement::Feature::kPowerForecastReporting,
+                    DeviceEnergyManagement::Feature::kStateForecastReporting, DeviceEnergyManagement::Feature::kStartTimeAdjustment,
+                    DeviceEnergyManagement::Feature::kPausable, DeviceEnergyManagement::Feature::kForecastAdjustment,
+                    DeviceEnergyManagement::Feature::kConstraintBasedAdjustment));
+            gDEMInstance->Init(); /* Register Attribute & Command handlers */
```

Comment:
if we assume gDEMDelegate alloc fails, why do we assume gDEMInstance always succeeds?

I would suggest to:
  - use unique_ptr
  - VerifyOrDie if allocation fails because that should not happen, and not try to handle out of memory in a main init.

Reply:
Done